3-0-java-core/3-6-4-random-field-comparator/src/main/java/com/bobocode/se/RandomFieldComparator.java

```diff
@@ -0,0 +1,82 @@
+package com.bobocode.se;
+
+import com.bobocode.util.ExerciseNotCompletedException;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Objects;
+import lombok.SneakyThrows;
+
+/**
+ * A generic comparator that is comparing a random field of the given class. The field is either primitive or
+ * {@link Comparable}. It is chosen during comparator instance creation and is used for all comparisons.
+ * <p>
+ * If no field is available to compare, the constructor throws {@link IllegalArgumentException}
+ *
+ * @param <T> the type of the objects that may be compared by this comparator
+ *<p><p>
+ *  <strong>TODO: to get the most out of your learning, <a href="https://www.bobocode.com/learn">visit our website</a></strong>
+ *  <p>
+ *
+ * @author Stanislav Zabramnyi
+ */
+public class RandomFieldComparator<T> implements Comparator<T> {
+
+    private final Class<T> targetType;
+    private final Field fieldToCompare;
+
+    public RandomFieldComparator(Class<T> targetType) {
+        Objects.requireNonNull(targetType);
+        this.targetType = targetType;
+        this.fieldToCompare = chooseFieldToCompare(targetType);
+    }
+
+    /**
+     * Compares two objects of the class T by the value of the field that was randomly chosen. It allows null values
+     * for the fields, and it treats null value grater than a non-null value.
+     *
+     * @param o1
+     * @param o2
+     * @return positive int in case of first parameter {@param o1} is greater than second one {@param o2},
+     *         zero if objects are equals,
+     *         negative int in case of first parameter {@param o1} is less than second one {@param o2}.
+     */
+    @SneakyThrows
+    @SuppressWarnings("unchecked")
+    @Override
+    public int compare(T o1, T o2) {
+        Objects.requireNonNull(o1);
+        Objects.requireNonNull(o2);
+        fieldToCompare.setAccessible(true);
+
```

Comment:
You user `Comparable` as raw type (without specifying a generic type), while you can actually provide a type argument and **use it property.** Both field values have the same type, so you need to create a helper method for compiler to catch that type. 

```java
@Override
    public int compare(T o1, T o2) {
        Objects.requireNonNull(o1);
        Objects.requireNonNull(o2);
        return compareFieldValues(o1, o2);
    }

    @SneakyThrows
    @SuppressWarnings("unchecked")
    private <U extends Comparable<? super U>> int compareFieldValues(T o1, T o2) {
        field.setAccessible(true);
        var value1 = (U) field.get(o1);
        var value2 = (U) field.get(o2);
        Comparator<U> comparator = Comparator.nullsLast(Comparator.naturalOrder());
        return comparator.compare(value1, value2);
    }
```

What I like about helper method is that you can put `@SneakyThrows` and well as `@SuppressWarnings` there, which makes method `compare` clean and simple. 🤷🏻‍♂️